packages/react-sdk/src/user-props/schema.ts

```diff
@@ -1,26 +1,46 @@
 import { z } from "zod";
 import { Asset } from "@webstudio-is/asset-uploader";
 
-const UserPropDb = z.object({
-  id: z.string(),
-  prop: z.string(),
-  value: z.union([z.string(), z.boolean(), z.number()]),
-  assetId: z.optional(z.string()),
-  required: z.optional(z.boolean()),
-});
-export type UserPropDb = z.infer<typeof UserPropDb>;
+// @todo move zod out of sdk
+// @todo clean db make objects .strict(),
+// as of now we already have asset and value simulateneously in db
+// and instead of strict error zod will ignore value
+const UserDbProp = z.union([
+  z.object({
+    id: z.string(),
+    prop: z.string(),
+    assetId: z.string(),
+    required: z.optional(z.boolean()),
+  }),
+  z.object({
+    id: z.string(),
+    prop: z.string(),
+    value: z.union([z.string(), z.boolean(), z.number()]),
+    required: z.optional(z.boolean()),
+  }),
+]);
 
-export const UserDbProps = z.array(UserPropDb);
-export type UserDbProps = z.infer<typeof UserDbProps>;
+export const UserDbProps = z.array(UserDbProp);
 
-export const UserProp = z.object({
-  id: z.string(),
-  prop: z.string(),
-  value: z.union([z.string(), z.boolean(), z.number()]),
-  asset: Asset.optional(),
-  required: z.optional(z.boolean()),
-});
-export type UserProp = z.infer<typeof UserProp>;
+export const UserDbPropsStrict = z.array(UserDbProp);
+
+export type UserDbProp = z.infer<typeof UserDbProp>;
+
+// @todo move zod out of sdk
+export const UserProp = z.union([
+  z.object({
+    id: z.string(),
+    prop: z.string(),
+    value: z.union([z.string(), z.boolean(), z.number()]),
+    required: z.optional(z.boolean()),
+  }),
+  z.object({
+    id: z.string(),
+    prop: z.string(),
+    asset: Asset,
+    required: z.optional(z.boolean()),
+  }),
+]);
```

Comment:
Why we should move Zod definition of data structures that defined in SDK and going to be part of SDKs public API?

But I agree something like `UserDbProp` should be moved